doc/debugger.rst

```diff
@@ -1773,6 +1811,161 @@ target endianness and interpret the bytecode based on that.
 .. note:: This command is somewhat incomplete at the moment and may be modified
    once the best way to do this in the debugger UI has been figured out.
 
+AppRequest request (0x22)
+-------------------------
+
+Format::
+
+    REQ <int: 0x22> [<tval>*] EOM
+    REP [<tval>*] EOM
+
+Example::
+
+    REQ 34 "GameInfo" "GetTitle" EOM
+    REP "Spectacles: Bruce's Story" EOM
+
+This is a custom request message whose meaning and semantics depend on the
+application.
+
+AppRequest messages are used for direct communication between the debug client
+and debug target over the Duktape debug protocol.  Both the meaning of a custom
+message and the dvalues it contains are entirely up to the implementation and
+depending on the needs of the application, need not be supported at all.
+
+See "Custom requests and notifications" below for more details.
+
+Custom requests and notifications
+=================================
+
+Starting in Duktape 1.5.x, Duktape supports direct communication between the
+debug client and debug target over the same transport by using the special
+AppRequest and AppNotify messages.  These messages have no meaning to Duktape,
+which merely serves to marshal them back and forth through a defined API.
+
+AppNotify messages may be sent by pushing the contents of the message to the
+stack and calling `duk_debugger_notify()` passing the number of values pushed.
+Each value pushed will be sent as a dvalue in the message.  So if you push two
+strings, "foo" and "bar", the client will see ``NFY 7 "foo" "bar" EOM``.
+
+AppRequest is used to make requests to the target which are not directly
+related to Ecmascript execution and may be implementation-dependant.  For
+example, an AppRequest might be used to:
+
+* Change the frame rate of a game engine
+
+* Reset/reboot an embedded target device while debugging
+
+* Perform or trigger software or script updates
+
+A target that wishes to support AppRequest should attach the debugger using the
+``duk_debugger_attach_custom()`` API and provide a request callback.  When an
+AppRequest is received, the request callback is invoked with the contents of
+the message on the value stack, and may push its own values to be sent in
+reply.
+
+This is a do-nothing request callback::
+
+    duk_idx_t duk_cb_debug_request(void* udata, duk_context* ctx, duk_idx_t nvalues) {
```

Comment:
The pointer notation doesn't match the usual custom, e.g. `void* udata` -> `void *udata`. Small trivia I know :)